Tests/BackendIntegrationTests/TrialOrIntroEligibilityCheckerIntegrationTests.swift

```diff
@@ -0,0 +1,104 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  TrialOrIntroEligibilityCheckerIntegrations.swift
+//
+//  Created by Nacho Soto on 4/6/22.
+
+import Nimble
+@testable import RevenueCat
+import StoreKitTest
+import XCTest
+
+// swiftlint:disable type_name
+
+class BaseTrialOrIntroEligibilityCheckerIntegrationTests: BaseStoreKitIntegrationTests {
+
+    private(set) var checker: TrialOrIntroPriceEligibilityChecker!
+
+    override func setUpWithError() throws {
+        try super.setUpWithError()
+
+        let currentUserProvider = MockCurrentUserProvider()
+
+        let systemInfo: SystemInfo = try .init(platformInfo: nil,
+                                               finishTransactions: false)
+        let backend: Backend = .init(httpClient: .init(systemInfo: systemInfo, eTagManager: .init()),
+                                     apiKey: Constants.apiKey,
+                                     attributionFetcher: .init(attributionFactory: .init(),
+                                                               systemInfo: systemInfo))
+        self.checker = .init(
+            receiptFetcher: .init(requestFetcher: .init(operationDispatcher: .default),
+                                  systemInfo: systemInfo),
+            introEligibilityCalculator: .init(productsManager: .init(systemInfo: systemInfo),
+                                              receiptParser: .init()),
+            backend: backend,
+            currentUserProvider: currentUserProvider,
+            operationDispatcher: .default,
+            productsManager: .init(systemInfo: systemInfo)
+        )
+    }
+
+    func checkEligibility(_ productIdentifiers: [String]) async throws -> [String: IntroEligibility] {
+        fatalError("This method must be overriden")
+    }
+
+    /// This test is similar to ``StoreKit1IntegrationTests/testEligibleForIntroBeforePurchaseAndIneligibleAfter``
+    /// except ran directly on `TrialOrIntroPriceEligibilityChecker`, which makes the test force
+    /// either the SK1 or SK2 implementation, so they can be tested independently.
+    final func runTestEligibleForIntroBeforePurchaseAndIneligibleAfter() async throws {
```

Comment:
Note that this function doesn't begin with `test`, so `XCTest` doesn't see it, and instead runs the tests in the subclasses (cc @taquitos)